core/domain/feedback_services.py

```diff
@@ -235,21 +296,54 @@ def update_messages_read_by_the_user(user_id, thread_id, message_ids):
     feedback_thread_user_model.put()
 
 
-def add_message_id_to_read_by_list(thread_id, user_id, message_id):
-    """Adds the message id to the list of message ids read by the user.
+def add_message_ids_to_read_by_list(thread_ids, user_id, message_ids):
+    """Adds the message ids to the list of message ids read by the user.
 
     Args:
-        thread_id: str. The id of the thread.
+        thread_ids: list(str). The ids of the threads.
         user_id: str. The id of the user reading the messages.
-        message_id: int. The id of the message.
-    """
-    feedback_thread_user_model = (
-        feedback_models.GeneralFeedbackThreadUserModel.get(
-            user_id, thread_id) or
-        feedback_models.GeneralFeedbackThreadUserModel.create(
-            user_id, thread_id))
-    feedback_thread_user_model.message_ids_read_by_user.append(message_id)
-    feedback_thread_user_model.put()
+        message_ids: list(str). The ids of the messages.
```

Comment:
Should state that the number of elements in this list should be the same as the number of elements in thread_ids, and check that condition at the outset of this function being run.